cmd/carbon-relay-ng/carbon-relay-ng.go

```diff
@@ -221,6 +221,11 @@ func main() {
 	}
 
 	select {}
+
```

Comment:
how is this supposed to work? i don't think the code will advance past line 223

Reply:
Not sure I understand this comment

Reply:
line 223 was `select {}` which blocks indefinitely and thus prevented the new code that @replay added from running. i've changed this to use a signal handler and trigger shutdown on certain signals